### ISSUE TITLE
Experiment with Playwright for E2E tests

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#6963]
Some developers in the core team have expressed frustration with Cypress. Another library from Microsoft recently gained some traction, overpassing Cypress in popularity (based on GitHub stars):

![image](https://github.com/opencollective/opencollective/assets/1556356/fef79a72-7ba9-4fd8-b4cd-df447612b681)
 
Compared to Cypress, Playwright is expected to be faster (it's easier to setup parallel tests) and easier to use. It would be interesting to setup a small experiment that adds Playwright to the codebase and translates one of the Cypress tests from `test/cypress` to see how it compares.

An interesting article on the topic: https://mtlynch.io/notes/cypress-vs-playwright/#parallel-tests-are-free-in-playwright

**Comment by VidhiRambhia** [comment]
I'm interested in picking this up. Can I work on this?

**Comment by Betree** [comment]
@VidhiRambhia Sure! Any proof of concept on this topic will be interesting for us.

**Comment by AtharvChandratre** [comment]
Hi @Betree, I would love to work on this with @VidhiRambhia!

**Comment by Betree** [comment]
@AtharvChandratre I'll let @VidhiRambhia reply on this, he's the first assigned but you two can work on it the way you prefer 🙂

**Comment by AtharvChandratre** [comment]
Hey @Betree, I was able to get some preliminary results:

Steps followed:

- Rewrote a Cypress Test File in Playwright (I chose 16-tiers-page.test.js) ([link to file](https://github.com/VidhiRambhia/opencollective-frontend/pull/1/files#diff-23591793ae8cc1ad15bae52467d43cbb74dc287604ca60c9488a225fa628f37a))
- Ran the Cypress test 10 times and recorded how much time each run took in Milliseconds ([link to edits made to the file to run it](https://github.com/VidhiRambhia/opencollective-frontend/pull/1/files#diff-bba38fdca49c6a7c97f2e719628b2c9ce8f0131870aed3e374a30778b94dc3ab))
- Ran the Playwright test 10 times ([link to the script used to run it](https://github.com/VidhiRambhia/opencollective-frontend/pull/1/files#diff-1422c62623a80fab953e7f5a1ee3a81e87fe1453065b8ca7086ffb7a93f2ddd7))

Results for the time taken (in milliseconds):
- Cypress:
  - Test 1 Average = 2124.9
  - Test 2 Average = 2110.4
- Playwright:
  - Test 1 Average = 1556.5
  - Test 2 Average = 1475.3

Playwright seems to be much faster than Cypress, with over a 30% decrease in execution time. 

Data for all runs:
- Cypress:
  - Test 1 time taken for runs = [2486, 2116, 2156, 2017, 2018, 2106, 2060, 1899, 2108, 2283]
  - Test 2 time taken for runs = [2018, 2120, 2037, 2031, 2135, 2074, 2292, 2225, 2141, 2031]
- Playwright:
  - Test 1 time taken for runs = [1483, 1402, 1395, 1426, 1570, 2019, 1873, 1443, 1516, 1438]
  - Test 2 time taken for runs = [1356, 1416, 1344, 1356, 1662, 1852, 1434, 1388, 1485, 1460]

**Comment by Betree** [comment]
@AtharvChandratre That's super interesting; thanks for these stats! Could you open a PR and link it, even if it's not ready yet, for us to see what it takes to get Playwright to work in this codebase?

**Comment by AtharvChandratre** [comment]
@Betree yep, here's the link to the PR - https://github.com/opencollective/opencollective-frontend/pull/9443

**Comment by Betree** [comment]
Current status: we've got a PR to demonstrate Playright here: https://github.com/opencollective/opencollective-frontend/pull/9799. It's fast, easy to work with, and the support for iframes is way more mature.

The main blocker right now stands with `pushRoute` not working at the end of the contribution flow, the interface is not updating. Clicking on a regular link works, and the URL does update so it looks like React is just not re-rendering properly.

We'll also need to check if opening the emails as we do with Cypress (on maildev) works before we can propose an RFC.

**Comment by Betree** [comment]
Playwright is not working on Fedora ATM: https://github.com/microsoft/playwright/issues/14736. Looks like we'll need to wait for it to mature.
